### PR TITLE
ENYO-5494: Fix minYear and maxYear in DatePicker for non-Gregorian locales

### PR DESCRIPTION
**Pull request by yeram**
### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Support DatePicker to calculate gregorian year to local year for minYear and maxYear props.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
DatePicker wasn't support gregorian date for minYear, maxYear.
It was supported in Enyo.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
ENYO-5494

### CODE REVIEW

File: packages/moonstone/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 The following is a curated list of changes in the Enact moonstone module, newest changes on the top.
 
+## [unreleased]
+
+### Fixed
+
+- `moonstone/DatePicker` to calculate min and max year in the current calender.
```
Comment (webOS101) [inline packages/moonstone/CHANGELOG.md]: Remove period from entry, please.

File: packages/moonstone/DatePicker/DatePicker.js [hunk 2]
```diff
@@ -63,7 +66,27 @@ const dateTimeConfig = {
 			.match(/([mdy]+)/ig)
 			.map(s => s[0].toLowerCase());
 
-		return {formatter, order};
+		/**
```
Comment (webOS101) [inline packages/moonstone/DatePicker/DatePicker.js]: This should not be a jsdoc comment block.  Comments can stay, but remove the 2nd `*` at the start.

File: packages/moonstone/DatePicker/DatePicker.js [hunk 3]
```diff
@@ -63,7 +66,27 @@ const dateTimeConfig = {
...
+		 * Converts a gregorian year to local year
+		 *
+		 * @param	{string}	year	gregorian year string
```
Comment (webOS101) [inline packages/moonstone/DatePicker/DatePicker.js]: Are these really strings?  The prop is a number.

File: packages/moonstone/CHANGELOG.md (hunk 1 above)
Comment (yeram) [reply packages/moonstone/CHANGELOG.md]: done!

File: packages/moonstone/DatePicker/DatePicker.js (hunk 2 above)
Comment (yeram) [reply packages/moonstone/DatePicker/DatePicker.js]: done!

File: packages/moonstone/DatePicker/DatePicker.js (hunk 3 above)
Comment (yeram) [reply packages/moonstone/DatePicker/DatePicker.js]: done!

**Comment by viodragon2** [comment]
Changed base to `release/1.13.4`

**Comment by ryanjduffy** [comment]
Noting that we have Travis failures due to the old code incompatibilities with new tooling/config.

**Pull request merged**